core/services/workflows/engine.go

```diff
@@ -217,7 +217,20 @@ func (e *Engine) loop(ctx context.Context) {
 				continue
 			}
 
-			err := e.startExecution(ctx, resp.Value)
+			te := &capabilities.TriggerEvent{}
+			err := resp.Value.UnwrapTo(te)
+			if err != nil {
+				e.logger.Errorf("could not unwrap trigger event", resp.Err)
+				continue
+			}
+
+			executionID, err := generateExecutionID(mockedWorkflowID, te.ID)
```

Comment:
Does this need to be hardcoded to mocked here? Can't we extract it from the actual workflow?